src/JsonSchema/Constraints/Constraint.php

```diff
@@ -30,6 +30,7 @@
     const CHECK_MODE_COERCE_TYPES =     0x00000004;
     const CHECK_MODE_APPLY_DEFAULTS =   0x00000008;
     const CHECK_MODE_EXCEPTIONS =       0x00000010;
+    const CHECK_MODE_DISABLE_FORMAT =   0x00000020;
```

Comment:
Normally, I'd agree with you. I would vastly prefer an on-by-default option.

The reason I did it this way here is because if you pass `$checkMode` directly, rather than going via `Factory::addConfig`, you'll clobber whatever the default is. In this case, that would mean a very high chance of users turning off `"format"` validation by accident (i.e. anybody who uses a custom `$checkMode` and didn't realise the implications of overwriting the default).

Thoughts?